query/src/sql/planner/binder/join.rs

```diff
@@ -49,13 +49,12 @@ impl<'a> Binder {
     ) -> Result<(SExpr, BindContext)> {
         let (left_child, left_context) =
             self.bind_table_reference(bind_context, &join.left).await?;
-        let (right_child, right_context) = self
-            .bind_table_reference(&left_context, &join.right)
-            .await?;
+        let (right_child, right_context) =
+            self.bind_table_reference(bind_context, &join.right).await?;
 
         check_duplicate_join_tables(&left_context, &right_context)?;
 
-        let mut bind_context = BindContext::new();
+        let mut bind_context = bind_context.replace();
```

Comment:
Seems you can directly use `with_parent`, then you don't need to think of a better function name to replace `replace`.
```rust
let mut bind_context = BindContext::with_parent(self.parent.clone());
```

Reply:
> Seems you can directly use `with_parent`, then you don't need to think of a better function name to replace `replace`.
> 
> ```rust
> let mut bind_context = BindContext::with_parent(self.parent.clone());
> ```

I was considering about that. The problem is `with_parent` takes a `Box<BindContext>` as input, while parent is a `Option<Box<BindContext>>`.